core/trino-server-rpm/src/test/java/io/trino/server/rpm/ServerIT.java

```diff
@@ -102,9 +102,10 @@ private static void testServer(String rpmHostPath, String expectedJavaVersion)
                 "yum localinstall -q -y " + rpm + "\n" +
                 // create Hive catalog file
                 "mkdir /etc/trino/catalog\n" +
+                "echo CONFIG_ENV[HMS_PORT]=9083 >> /etc/trino/env.sh\n" +
                 "cat > /etc/trino/catalog/hive.properties <<\"EOT\"\n" +
                 "connector.name=hive-hadoop2\n" +
-                "hive.metastore.uri=thrift://localhost:9083\n" +
+                "hive.metastore.uri=thrift://localhost:${ENV:HMS_PORT}\n" +
```

Comment:
looks like local testing modification

it would be nice to have a test for that, but let's make it a dedicated one

Reply:
It's not a local testing modification. It extends the test coverage to ensure envs are passed properly. This is an integration test, not a unit test. I'm concerned that a dedicated test would increase the total duration. Are you sure it's worth putting in a separate test?

Reply:
How much time we would add and how much time the job already takes today?

BTW the test is not very directly (there is no explicit assertion, so it could _in theory_ become ignored by the server.)
Can we inject eg `node.environment` and verify what's the actual value of that?

Reply:
Trino throws an error when configuration files reference an unset env var.

Reply:
These two tests in ServerIT run in the pipeline in slightly over a minute:
```
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 70.948 s - in io.trino.server.rpm.ServerIT
```
so I'd expect a separate test to add ~30 seconds.

Reply:
And the job total time is 20 minute. I think it's reasonable to have this as a separate test, but i think i am OK keeping this inline.

would it be possible to add an explicit assertion for the feature?

Reply:
I added a custom node id and an assert for it.